File: examples/python/getting_started.py
"""
Getting started with the Glider API.

This script connects to a Modos e-ink display and sets the entire screen to a
fast monochrome mode. Run it first to confirm your setup is working.

Prerequisites:
  pip install .                    # build and install the Python binding
  sudo chmod 0666 /dev/hidraw*     # Linux: grant access to the HID device
                                   # (see README for a permanent udev rule)
"""

from glider_api import Display, DisplayConfig, Mode

config = DisplayConfig.glider_standard()

try:
    display = Display.new_with_config(config)
except TypeError as e:
    print(f"Could not connect to display: {e}")
    print("Check that the Glider is plugged in and you have permission to")
    print("access the HID device (see README for Linux udev setup).")
    raise SystemExit(1)

display.set_mode(Mode.FastMonoNoDither, config.full_screen())
print("Done — the display should now be in fast monochrome mode.")

File: examples/rust/getting_started.rs
//! Getting started with the Glider API.
//!
//! Connects to a Modos e-ink display and sets the entire screen to fast
//! monochrome mode. Run this first to confirm your setup is working.
//!
//! ```
//! cargo run --example getting_started
//! ```

use glider_api::{Display, DisplayConfig, Mode};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let config = DisplayConfig::glider_standard();

    let display = Display::new_with_config(&config).map_err(|e| {
        eprintln!("Could not connect to display: {e}");
        eprintln!("Check that the Glider is plugged in and you have permission");
        eprintln!("to access the HID device (see README for Linux udev setup).");
        e
    })?;

    display.set_mode(&Mode::FastMonoNoDither, &config.full_screen())?;
    println!("Done — the display is now in fast monochrome mode.");
    Ok(())
}
